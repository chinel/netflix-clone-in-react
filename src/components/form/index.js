import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
