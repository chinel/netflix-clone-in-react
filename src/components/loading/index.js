import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
