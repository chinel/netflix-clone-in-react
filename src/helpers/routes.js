import React from "react";
import {Route, Redirect} from "react-router-dom";

export function IsUserRedirect ({user, loggedInPath, children, ...restProps}) {
    return (
        <Route 
        {...restProps}
         render={() => {
             if(!user){
                 return children;
             }

             if(user){
                 return <Redirect
                 to={{
                     pathname: loggedInPath
                 }}
                 />
             }
            return null;
            }
        
        }
        />
    )
}

