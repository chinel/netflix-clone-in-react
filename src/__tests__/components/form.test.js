import React from "react";
import { render } from "@testing-library/react";
import { Form } from "../../components";
import * as ROUTES from "../../constants/routes";

jest.mock("react-router-dom", () => ({
  Link: "Link",
}));

describe("<Form />", () => {
  it("renders the <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email Address" onChange={() => {}} />
          <Form.Input
            placeholder="Password"
            type="password"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            Sign In
          </Form.Submit>
        </Form.Base>

        <Form.Text>
          New to Netflix?{" "}
          <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCaptcha to ensure that you're not a
          bot. Learn more.
        </Form.TextSmall>
      </Form>
    );

    expect(
      getByText(
        "This page is protected by Google reCaptcha to ensure that you're not a bot. Learn more."
      )
    ).toBeTruthy();
  });
});
