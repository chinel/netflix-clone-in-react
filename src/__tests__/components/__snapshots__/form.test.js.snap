// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1ohh9u9-0 iZnuyL"
>
  <h1
    class="form__Title-sc-1ohh9u9-3 cyHSTB"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1ohh9u9-2 hSfVbW"
  >
    <input
      class="form__Input-sc-1ohh9u9-7 hXqiVp"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-1ohh9u9-7 hXqiVp"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1ohh9u9-8 kvFzBG"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-1ohh9u9-4 cJqEow"
  >
    New to Netflix?
     
    <link
      class="form__TextLink-sc-1ohh9u9-6 eCwtFN"
      to="/signup"
    >
       
      Sign up now.
       
    </link>
  </p>
  <p
    class="form__TextSmall-sc-1ohh9u9-5 jcWIxY"
  >
    This page is protected by Google reCaptcha to ensure that you're not a bot. Learn more.
  </p>
</div>
`;
